pandas/io/parquet.py

```diff
@@ -154,6 +155,32 @@ def __init__(self):
 
         self.api = pyarrow
 
+    @staticmethod
+    def _write_attrs(table, df: DataFrame):
```

Comment:
make this a free-function, type the output and add a doc-string

Reply:
The reason for it being a static method is to ensure it is only used with `PyArrowImpl` and there is no confusion with `FastParquetImpl`. Since you would prefer them to be free functions, how about naming them 
`_pyarrow_write_attrs` / `_pyarrow_read_attrs`?

Reply:
that's fine